source/parser.hera

```diff
@@ -1577,16 +1609,29 @@ AmpersandUnaryPrefix
   [!~+-]+
 
 ThinArrowFunction
-  Parameters:parameters ReturnTypeSuffix?:suffix _* Arrow:arrow BracedOrEmptyBlock:block ->
+  ( Async _ )?:async Parameters:parameters ReturnTypeSuffix?:suffix _* Arrow:arrow BracedOrEmptyBlock:block ->
+    if (hasAwait(block) && !async) {
+      async = "async "
+    }
+
+    let generator = ""
```

Comment:
Maybe `undefined` (`let generator`) would be better here? It would correspond to the `Star?` rule.